lemur/plugins/lemur_acme/ultradns.py

```diff
@@ -0,0 +1,323 @@
+            # and a zone for b.c.com, we want to use b.c.com.
+            if z.count(".") > zone_name.count("."):
+                zone_name = z
+    if not zone_name:
+        metrics.send("ultradns_no_zone_name", "counter", 1)
+        raise Exception("No UltraDNS zone found for domain: {}".format(domain))
+    return zone_name
+
+
+def create_txt_record(domain, token, account_number):
+    # Create a TXT record for the given domain.
+    # The part of the domain that matches with the zone becomes the zone name.
+    # The remainder becomes the owner name (referred to as node name here)
+    # Example: Let's say we have a zone named "exmaple.com" in UltraDNS and we
+    # get a request to create a cert for lemur.example.com
+    # Domain - _acme-challenge.lemur.example.com
+    # Matching zone - example.com
+    # Owner name - _acme-challenge.lemur
+
+    zone_name = get_zone_name(domain, account_number)
+    zone_parts = len(zone_name.split("."))
+    node_name = ".".join(domain.split(".")[:-zone_parts])
+    fqdn = "{0}.{1}".format(node_name, zone_name)
+    path = "/v2/zones/{0}/rrsets/TXT/{1}".format(zone_name, node_name)
+    params = {
+        "ttl": 5,
+        "rdata": [
+            "{}".format(token)
+        ],
+    }
+
+    try:
+        _post(path, params)
+        current_app.logger.debug(
+            "TXT record created: {0}, token: {1}".format(fqdn, token)
+        )
+    except Exception as e:
+        current_app.logger.debug(
+            "Unable to add record. Domain: {}. Token: {}. "
+            "Record already exists: {}".format(domain, token, e),
+            exc_info=True,
+        )
+
+    change_id = (fqdn, token)
+    return change_id
+
+
+def delete_txt_record(change_id, account_number, domain, token):
+    # Delete the TXT record that was created in the create_txt_record() function.
+    # UltraDNS handles records differently compared to Dyn. It creates an RRSet
+    # which is a set of records of the same type and owner. This means
+    # that while deleting the record, we cannot delete any individual record from
+    # the RRSet. Instead, we have to delete the entire RRSet. If multiple certs are
+    # being created for the same domain at the same time, the challenge TXT records
+    # that are created will be added under the same RRSet. If the RRSet had more
+    # than 1 record, then we create a new RRSet on UltraDNS minus the record that
+    # has to be deleted.
+
+    if not domain:
+        current_app.logger.debug("delete_txt_record: No domain passed")
+        return
+
+    zone_name = get_zone_name(domain, account_number)
+    zone_parts = len(zone_name.split("."))
+    node_name = ".".join(domain.split(".")[:-zone_parts])
+    path = "/v2/zones/{}/rrsets/16/{}".format(zone_name, node_name)
+
+    try:
+        rrsets = _get(path)
+        record = Record(rrsets)
+    except Exception as e:
+        metrics.send("delete_txt_record_geterror", "counter", 1)
```

Comment:
recommend `metrics.send(f"{function}.geterror,...`

Making the metrics based on the function name will allow us to ensure consistency.
Prefer dot notation to make it easier to distinguish function namespace against the metric name, underscores can be confusing because the function name also contains underscores